pennylane/data/data_manager/__init__.py

```diff
@@ -0,0 +1,322 @@
+
+    return [Dataset.open(Path(dest_path), "r") for dest_path in dest_paths]
+
+
+def list_datasets() -> dict:
+    r"""Returns a dictionary of the available datasets.
+
+    Return:
+        dict: Nested dictionary representing the directory structure of the hosted datasets.
+
+    **Example:**
+
+    Note that the results of calling this function may differ from this example as more datasets
+    are added. For updates on available data see the `datasets website <https://pennylane.ai/qml/datasets.html>`_.
+
+    .. code-block :: pycon
+
+        >>> qml.data.list_datasets()
+        {'qchem': {'H2': {'6-31G': ['0.5', '0.54', '0.58', ... '2.02', '2.06', '2.1'],
+                          'STO-3G': ['0.5', '0.54', '0.58', ... '2.02', '2.06', '2.1']},
+                   'HeH+': {'6-31G': ['0.5', '0.54', '0.58', ... '2.02', '2.06', '2.1'],
+                            'STO-3G': ['0.5', '0.54', '0.58', ... '2.02', '2.06', '2.1']},
+                   'LiH': {'STO-3G': ['0.5', '0.54', '0.58', ... '2.02', '2.06', '2.1']},
+                   'OH-': {'STO-3G': ['0.5', '0.54', '0.58', ... '0.94', '0.98', '1.02']}},
+        'qspin': {'Heisenberg': {'closed': {'chain': ['1x16', '1x4', '1x8'],
+                                            'rectangular': ['2x2', '2x4', '2x8', '4x4']},
+                                 'open': {'chain': ['1x16', '1x4', '1x8'],
+                                        'rectangular': ['2x2', '2x4', '2x8', '4x4']}},
+                  'Ising': {'closed': {'chain': ['1x16', '1x4', '1x8'],
+                                        'rectangular': ['2x2', '2x4', '2x8', '4x4']},
+                            'open': {'chain': ['1x16', '1x4', '1x8'],
+                                    'rectangular': ['2x2', '2x4', '2x8', '4x4']}}}}
+    """
+
+    def remove_paths(foldermap):
+        """Copies the foldermap, converting the bottom-level mapping of parameters
+        to Paths to a list of the parameters."""
+        value = next(iter(foldermap.values()))
+
+        if not isinstance(value, typing.Mapping):
+            return sorted(foldermap.keys())
+
+        return {param: remove_paths(foldermap[param]) for param in foldermap.keys()}
+
+    return remove_paths(_get_foldermap())
+
+
+def list_attributes(data_name):
+    r"""List the attributes that exist for a specific ``data_name``.
+
+    Args:
+        data_name (str): The type of the desired data
+
+    Returns:
+        list (str): A list of accepted attributes for a given data name
+    """
+    data_struct = _get_data_struct()
+    if data_name not in data_struct:
+        raise ValueError(
+            f"Currently the hosted datasets are of types: {list(data_struct)}, but got {data_name}."
+        )
+    return data_struct[data_name]["attributes"]
+
+
+def _interactive_request_attributes(options):
+    """Prompt the user to select a list of attributes."""
+    prompt = "Please select attributes:"
+    for i, option in enumerate(options):
+        if option == "full":
```

Comment:
Now that I think about it, I think we should only mention `"full"` to users to keep both `interactive` and `load` equivalent for users.